src/_guides/openapi/code-first.md

```diff
@@ -0,0 +1,177 @@
+
+The main concern with API Design-first is "drift", where the code and schema diverge over time. The annotations approach only pretended to solve this problem, confusing proximity with accuracy. The comments above code could still completely fail to accurately describe the code below, but nobody would ever notice until a user complained about it. 
+
+The OpenAPI-aware Framework approach does solve this by making the code a single source of truth, but the Design-first approach can be used to make any framework OpenAPI aware, with a source of truth that exists before the code, and continues to be useful after the code is built.
+
+Server-side validation with OpenAPI can avoid the need to write lots of request validation, using middleware to compare incoming HTTP requests against the contract defined in the OpenAPI description and automatically return validation errors instead of having to build that all our yourself.
+
+- **Java**
+	- [openapi-request-response-validation](https://github.com/gcatanese/openapi-request-response-validation)
+- **JavaScript**
+	- [express-openapi-validator](https://www.npmjs.com/package/express-openapi-validator)
+	- [fastify-openapi-glue](https://www.npmjs.com/package/fastify-openapi-glue)
+	- [openapi-enforcer](https://www.npmjs.com/package/openapi-enforcer)
+	- [openapi-validator-middleware](https://www.npmjs.com/package/openapi-validator-middleware)
+- **Perl**
+	- [JSONSchema Validator](https://metacpan.org/pod/JSONSchema::Validator)
+- **PHP**
+	- [openapi-psr7-validator](https://github.com/thephpleague/openapi-psr7-validator)
+- **Ruby**
+	- [committee](https://rubygems.org/gems/committee)
+
+Responses can be validated using any existing test suite, with all popular testing frameworks supporting OpenAPI by extension. 
+
+- **JavaScript** 
+	- [chai-openapi-response-validator](https://www.npmjs.com/package/chai-openapi-response-validator)
+	- [jest-openapi](https://www.npmjs.com/package/jest-openapi)
+- **PHP**
+	- [Spectator](https://github.com/hotmeteor/spectator)
+- **Ruby**
+	- [rspec-openapi](https://github.com/exoego/rspec-openapi)
+
+Years ago the API Design-first workflow was a rough approach, but thankfully a whole bunch of tooling developers spent those years making things excellent, and now it's easier than ever. Bump.sh adds to that legacy by adding amazing [change detection](/help/changes-management/), helping check the OpenAPI in your git repository for changes that would be breaking for end users, letting you know in the pull request when there's a problem, providing beyond a shadow of a doubt that having your OpenAPI as a source of truth in a git repository along with your source code is not only handy, but probably the best way to go for many teams. 
```

Comment:
:heart: 

Should we link to the [help page documenting the GH action for diffs on PRs](https://docs.bump.sh/help/continuous-integration/github-actions/#diff-on-pull-requests-only)?